test/test_autograd.py

```diff
@@ -2165,6 +2165,10 @@ def test_dir(self):
         keys = dir(x)
         self.assertIn('shape', keys)
 
+        # __cuda_array_interface__ attribute is not valid for non-cuda tensors,
+        # Marks this via "AttributeError", hasattr is False
+        keys.remove("__cuda_array_interface__")
```

Comment:
Can we fix `dir` to not return it for CPU tensors then? This test is there for a reason